lib/requireAccess.js

```diff
@@ -19,12 +19,11 @@ module.exports = function(options) {
     options.adminToken = res.body.firebaseToken;
     return;
   }).catch(function(err) {
-    if (err && err.exit) {
-      return RSVP.reject(err);
+    var errCode = _.get(err, 'context.body.error.code');
+    if (errCode === 'AUTHENTICATION_REQUIRED') {
```

Comment:
Correct, the `firebase deploy` command [calls `requireAccess()`](https://github.com/firebase/firebase-tools/blob/98dee4b3e6c29bfd058728e259325d410b6f4641/commands/deploy.js#L15) which [calls `requireAuth()`](https://github.com/firebase/firebase-tools/blob/98dee4b3e6c29bfd058728e259325d410b6f4641/lib/requireAccess.js#L15) which [creates a new `FirebaseError` without this error code](https://github.com/firebase/firebase-tools/blob/98dee4b3e6c29bfd058728e259325d410b6f4641/lib/requireAuth.js#L25).

If you are logged out, this is what you see:

```
$ firebase deploy

Error: Command requires authentication, please run firebase login
```